src/Microsoft.OData.Core/Evaluation/ODataConventionalIdMetadataBuilder.cs

```diff
@@ -296,7 +298,10 @@ private Uri GetContainingEntitySetUri(Uri baseUri, ODataUri odataUri)
             }
 
             // trim all the un-needed segments 
-            segments = segments.GetRange(0, lastIndex);
+            if (lastIndex >= 0)
+            {
+                segments.RemoveAt(lastIndex);
+            }

```

Comment:
Would calling `segments.RemoveRange(lastIndex, countToRemove)` to trim the un-needed segments instead of calling `RemoveAt` once at a time be a better alternative?

Reply:
+1 -- it seems like the existing code would be more performant than individually removing elements.

---
In reply to: [570827746](https://github.com/OData/odata.net/pull/1993#discussion_r570827746) [](ancestors = 570827746)

Reply:
@Sreejithpin just wanted to drop these two here as they may provide some more context as to why this code exists here as they contain the details and comments https://github.com/OData/odata.net/issues/1725 & the pr https://github.com/OData/odata.net/pull/1854.